dataset.proto

```diff
@@ -34,9 +40,11 @@ message DataSet {
     string url = 1;
   }
 
-  Categorization categorization = 1;
-  AdFraud        adfraud        = 2;
-  // 3 and 4 are reserved
-  Malicious      malicious      = 5;
-  Echo           echo           = 6;
+  Categorization         categorization = 1;
+  AdFraud                adfraud        = 2;
+  // 3 is reserved
+  // if SENTIMENT is not requested, keywords will map to zero values  
+  map<string, Sentiment> keyword        = 4;
```

Comment:
I think it would be cleaner (especially for the sake of the reflect based code in dataset.go and elsewhere) to have two fields, keyword and sentiment. This has a few advantages.

1. By keeping the number of types and fields equal, it allows the reflect code to go unchanged, without needing to account for special cases
2. It can hide the notion of sentiment from keyword only users who have no access/need for sentiment, and as such will have to write code to actively ignore the extra field
3. As is, if we're trying to extract sentiment values from titan-dataset, we would have to check keyword, iterate through every member of the map and check to see if it's neutral, to deduce whether or not sentiment had in fact been processed, or if it was only a keyword request. 

One catch is that there is the risk of duplicating data, but I think it can be managed. There should be no use case in which both keyword AND sentiment are selected, at least from our internal query standpoint (zvelo-api can check for this, and change the request_datasets to just sentiment), therefore both fields shouldn't ever be filled, thus no duplication. 

The other is that if we have computed the sentiment of a URL, and a subsequent keyword request for that URL comes in, we will have to handle the special case of copying the keywords from the stored sentiment field into the keyword field. This is handling a special case, but it is limited to one place, instead of many.